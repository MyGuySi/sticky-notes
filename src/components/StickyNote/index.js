// @flow

import { type NoteColor } from '../../types'
import * as React from 'react';
import styles from './styles.module.css';

type Props = {
  color: NoteColor,
  children: React.Node
}

function StickyNote (props: Props) {
  const className = styles.note + ' ' + styles[props.color];
  return (
    <div className={className}>
      {props.children}
    </div>
  )
}

export default StickyNote;