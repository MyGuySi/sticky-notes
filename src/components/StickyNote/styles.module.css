:root {
  --color-note: white;
  --color-note-shadow: rgba(0, 0, 0, 0.1);
}

.note {
  padding: 1.5rem;
  border-radius: 0.125rem;
  background-color: var(--color-note);
  box-shadow: 0 8px 16px var(--color-note-shadow);
}

.pink {
  --color-note: #ff65a3;
  --color-note-shadow: rgba(175, 0, 71, 0.2);
}

.cyan {
  --color-note: #7afcff;
  --color-note-shadow: rgba(0, 150, 150, 0.2);
}

.yellow {
  --color-note: #fff740;
  --color-note-shadow: rgba(168, 159, 0, 0.2);
}