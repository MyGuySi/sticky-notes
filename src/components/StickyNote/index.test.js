// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import StickyNote from './index'

test('renders correctly', () => {
  const component = renderer.create(
    <StickyNote color='pink'>
      <p>Test</p>
    </StickyNote>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
