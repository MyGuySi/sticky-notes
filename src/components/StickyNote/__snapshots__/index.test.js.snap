// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="note pink"
>
  <p>
    Test
  </p>
</div>
`;
